cedar-policy-core/src/est/head_constraints.rs

```diff
@@ -201,6 +201,131 @@ impl ActionConstraint {
     }
 }
 
+impl std::fmt::Display for PrincipalConstraint {
```

Comment:
To reduce code duplication, maybe these could share a function that takes `all_str`, `eq_str` and `in_str` as args?

Reply:
Not quite sure what you mean.  There are more different types here than you might think at first glance (for instance, for `PrincipalConstraint`, the `In` variant contains `PrincipalOrResourceInConstraint`, while for `ActionConstraint`, the `In` variant contains `ActionInConstraint`)

Reply:
Makes sense.